sonar-orchestrator/src/test/java/com/sonar/orchestrator/OrchestratorBuilderTest.java

```diff
@@ -237,42 +237,89 @@ public void use_default_admin_credentials_for_builds_if_enabled() {
+      orchestrator.executeBuildQuietly(fakeBuild);
+      verifyTokenFromProperty(fakeBuild, "sonar.token", "custom-property-token");
 
       fakeBuild = FakeBuild.create(successResult);
       fakeBuild.addArgument("-Dsonar.login=custom-argument-token");
       orchestrator.executeBuildQuietly(fakeBuild);
+      verifyTokenFromArgument(fakeBuild, "sonar.login", "custom-argument-token");
 
-      assertThat(fakeBuild.arguments()).contains("-Dsonar.login=custom-argument-token");
+      fakeBuild = FakeBuild.create(successResult);
+      fakeBuild.addArgument("-Dsonar.token=custom-argument-token");
+      orchestrator.executeBuildQuietly(fakeBuild);
+      verifyTokenFromArgument(fakeBuild, "sonar.token", "custom-argument-token");
 
       Build<FakeBuild> fakeBuild1 = FakeBuild.create(successResult);
       Build<FakeBuild> fakeBuild2 = FakeBuild.create(successResult);
       Build<FakeBuild> fakeBuild3 = FakeBuild.create(successResult);
       orchestrator.executeBuilds(fakeBuild1, fakeBuild2, fakeBuild3);
 
-      assertThat(fakeBuild1.getProperties()).hasEntrySatisfying("sonar.login", s -> assertThat(s).isNotBlank());
-      assertThat(fakeBuild2.getProperties()).hasEntrySatisfying("sonar.login", s -> assertThat(s).isNotBlank());
-      assertThat(fakeBuild3.getProperties()).hasEntrySatisfying("sonar.login", s -> assertThat(s).isNotBlank());
+      verifyTokenCreated(fakeBuild1, "sonar.token");
+      verifyTokenCreated(fakeBuild2, "sonar.token");
+      verifyTokenCreated(fakeBuild3, "sonar.token");
+    } finally {
+      orchestrator.stop();
+    }
+  }
+
+  @Test
+  public void executeBuild_whenDefaultAdminCredentialsEnabledOnOldVersion_shouldUseLoginProperty() {
+    Orchestrator orchestrator = new OrchestratorBuilder(Configuration.create())
+      .setSonarVersion(LTS_ALIAS)
+      .useDefaultAdminCredentialsForBuilds(true)
+      .build();
+
+    try {
+      orchestrator.start();
 
+      BuildResult successResult = mock(BuildResult.class);
+      when(successResult.isSuccess()).thenReturn(true);
+
+      Build<FakeBuild> fakeBuild = FakeBuild.create(successResult);
+      orchestrator.executeBuild(fakeBuild);
+      String adminToken = verifyTokenCreated(fakeBuild, "sonar.login");
+
+      fakeBuild = FakeBuild.create(successResult);
+      orchestrator.executeBuild(fakeBuild);
+      verifyTokenReused(fakeBuild, "sonar.login", adminToken);
     } finally {
       orchestrator.stop();
     }
   }
 
+  private String verifyTokenCreated(Build<FakeBuild> fakeBuild, String property) {
+    String token = fakeBuild.getProperties().get(property);
+    assertThat(token).isNotBlank();
+    return token;
+  }
+
+  private void verifyTokenReused(Build<FakeBuild> fakeBuild, String property, String tokenToReuse) {
+    String token = fakeBuild.getProperties().get(property);
+    assertThat(token).isNotBlank().isEqualTo(tokenToReuse);
+  }
+
+  private void verifyTokenFromProperty(Build<FakeBuild> fakeBuild, String property, String value) {
```

Comment:
I have Intellij warning on this: "Actual value of parameter 'value' is always 'custom-property-token"
We can get rid of the last parameter